### PR TITLE
Add mitxonline_username and edxor_username to MITx combined user

### PR DESCRIPTION
**Pull request by rachellougee**
## Description

This PR dedupe users exist on both MITx Online and edx.org based on their linked profile from MicroMasters. But there are many edxorg usernames on MM that don't actually exist on edxorg, for those users, is_edxorg_user = false with their MITx Online username in this table
   - is_mitxonline_user
   - is_edxorg_user
   - user_mitxonline_id
   -  user_edxorg_id
   - user_mitxonline_username
   - user_edxorg_username
   - user_mitxonline_email
   -  user_edxorg_email

For MITx Online user, is_mitxonline_user =true. For the edx.org user, is_edxorg_user = true. For users exist on both platforms, is_mitxonline_user and  is_edxorg_user both are true

## Motivation and Context

No issue was created. This was from our data platform meeting today

## How Has This Been Tested?

 `dbt run --select int__mitx__users  --var 'schema_suffix: rlougee' --target dev_production`
 `dbt test --select int__mitx__users  --var 'schema_suffix: rlougee' --target dev_production`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql [hunk 1]
```diff
@@ -10,33 +12,197 @@ with mitxonline_users as (
...
+        , user_full_name
+        , user_address_country
+        , user_highest_education
+        , user_gender
+        , user_birth_year
+    from mitx_users_dups
+    where row_num = 2
+)
+
+, mitx_users_dups_combined as (
```
Comment (abeglova) [inline src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:149]: So this query works but it is a little complicated could you do a full outer join from mitxonline_users to edxorg_users? Or am I missing something?

**Review by abeglova — CHANGES_REQUESTED** [review]

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql (hunk 1 above)
Comment (rachellougee) [reply src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:149]: This query is to combine user profiles IF users exist on MITx Online and edX.org, INNER JOIN here does the trick, not sure what we try to do by changing to full outer joins. Let me know if I misunderstand

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql [hunk 2]
```diff
@@ -10,33 +12,67 @@ with mitxonline_users as (
...
+        , edxorg_users.user_email as user_edxorg_email
+        , edxorg_users.user_full_name
+        , edxorg_users.user_country as user_address_country
+        , edxorg_users.user_highest_education
+        , edxorg_users.user_gender
+        , edxorg_users.user_birth_year
+        , true as is_edxorg_user
+        , coalesce(mitxonline_users.user_username is not null, false) as is_mitxonline_user
+    from edxorg_users
+    left join mitxonline_users on mitxonline_users.user_edxorg_username = edxorg_users.user_username
```
Comment (abeglova) [inline src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:50]: I don't think you need the left join anymore here or in line 31

Comment (rachellougee) [reply src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql]: Done. Removed left joins

**Review by abeglova — APPROVED** [review]
lgtm

**Pull request merged**